homeassistant/components/homewizard/sensor.py

```diff
@@ -308,39 +365,33 @@ async def async_setup_entry(
     """Initialize sensors."""
     coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
 
-    entities: list[HWEnergySensor] = []
-    if coordinator.data.data is not None:
-        entities.extend(
-            HWEnergySensor(coordinator, entry, description)
-            for description in SENSORS
-            if getattr(coordinator.data.data, description.key) is not None
-        )
-    async_add_entities(entities)
+    async_add_entities(
```

Comment:
This removes check for `coordinator.data.data is not None`
Should it be kept with an early exit?
```suggestion
    if coordinator.data.data is None:
        return
    async_add_entities(
```

Edit: I see that the test has been adjusted around this, so I assume it's on purpose...

Reply:
> This removes check for coordinator.data.data is not None
> Should it be kept with an early exit?

Correct, `data` can be `None`. Typing of that has been adjusted in previous PRs. This is merely cleaning up an leftover.